From: Peter Zijlstra <peterz@infradead.org>
Date: Tue, 8 Mar 2022 16:30:47 +0100
Subject: x86: Mark stop_this_cpu() __noreturn
Git-commit: f9cdf7ca57cada055f61ef6d0eb4db21c3f200db
Patch-mainline: v5.18-rc1
References: git-fixes

vmlinux.o: warning: objtool: smp_stop_nmi_callback()+0x2b: unreachable instruction

0000 0000000000047cf0 <smp_stop_nmi_callback>:
...
0026    47d16:  e8 00 00 00 00          call   47d1b <smp_stop_nmi_callback+0x2b>       47d17: R_X86_64_PLT32   stop_this_cpu-0x4
002b    47d1b:  b8 01 00 00 00          mov    $0x1,%eax

Signed-off-by: Peter Zijlstra (Intel) <peterz@infradead.org>
Acked-by: Josh Poimboeuf <jpoimboe@redhat.com>
Link: https://lore.kernel.org/r/20220308154319.290905453@infradead.org

Acked-by: Nikolay Borisov <nik.borisov@suse.com>
---
 arch/x86/include/asm/processor.h |    2 +-
 arch/x86/kernel/process.c        |    2 +-
 tools/objtool/check.c            |    1 +
 3 files changed, 3 insertions(+), 2 deletions(-)

--- a/arch/x86/include/asm/processor.h
+++ b/arch/x86/include/asm/processor.h
@@ -970,7 +970,7 @@ bool xen_set_default_idle(void);
 #define xen_set_default_idle 0
 #endif
 
-void stop_this_cpu(void *dummy);
+void __noreturn stop_this_cpu(void *dummy);
 void df_debug(struct pt_regs *regs, long error_code);
 void microcode_check(void);
 
--- a/arch/x86/kernel/process.c
+++ b/arch/x86/kernel/process.c
@@ -575,7 +575,7 @@ bool xen_set_default_idle(void)
 }
 #endif
 
-void stop_this_cpu(void *dummy)
+void __noreturn stop_this_cpu(void *dummy)
 {
 	local_irq_disable();
 	/*
--- a/tools/objtool/check.c
+++ b/tools/objtool/check.c
@@ -174,6 +174,7 @@ static int __dead_end_function(struct ob
 		"fortify_panic",
 		"rewind_stack_do_exit",
 		"cpu_bringup_and_idle",
+		"stop_this_cpu",
 	};
 
 	if (func->bind == STB_WEAK)
